test/config/mocks/injector.js

```diff
@@ -0,0 +1,7 @@
+angular.module('esn.test.injector', []).run(function($q) {
+  window.$q = $q;
+});
+
+beforeEach(angular.mock.module('esn.test.injector'));
+
+window.$q = angular.injector(["ng"]).get("$q");
```

Comment:
Does this definition apply for all tests?

```suggestion
window.$q = angular.injector(['ng']).get('$q');
```

Reply:
Yes, it does apply for all tests.

Reply:
So you do not need to repeat it in test cases?

Reply:
No, we don't. I'll remove the redundant definition of `$q` in the test cases.

Reply:
Fixed